sdk/config.go

```diff
@@ -0,0 +1,124 @@
+package sdk
+
+import (
+	"os"
+	"strconv"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/sdk/trace"
+)
+
+type HeliosConfig struct {
+	serviceName       string
+	apiToken          string
+	sampler           trace.Sampler
+	collectorEndpoint string
+	collectorPath     string
+	environment       string
+	commitHash        string
+}
+
+// Keys and their matching env vars
+const samplingRatioKey = "samplingRatio"
+const samplingRatioEnvVar = "HS_SAMPLING_RATIO"
+const environmentKey = "environment"
+const environmentEnvVar = "HS_ENVIRONMENT"
+const collectorEndpointKey = "collectorEndpoint"
+const collectorEndpointEnvVar = "HS_COLLECTOR_ENDPOINT"
+const collectorPathKey = "collectorPath"
+const collectorPathEnvVar = "HS_COLLECTOR_PATH"
+const commitHashKey = "commitHash"
+const commitHashEnvVar = "HS_COMMIT_HASH"
+
+// Default values
+const defaultCollectorEndpoint = "collector.heliosphere.io:443"
+const defaultCollectorPath = "traces"
+
+func getConfigByKey(key string, attrs []attribute.KeyValue) attribute.KeyValue {
+	for i := range attrs {
+		if string(attrs[i].Key) == key {
+			return attrs[i]
+		}
+	}
+
+	return attribute.KeyValue{Key: "", Value: attribute.Value{}}
+}
+
+func getSampler(samplerConfig attribute.KeyValue) trace.Sampler {
+	samplingRatio := os.Getenv(samplingRatioEnvVar)
+	if samplingRatio != "" {
+		res, err := strconv.ParseFloat(samplingRatio, 64)
+		if err != nil {
+			return trace.AlwaysSample()
+		}
+
+		return trace.TraceIDRatioBased(res)
+	}
+
+	if samplerConfig.Key == "" {
+		return trace.AlwaysSample()
+	}
+
+	return trace.TraceIDRatioBased(samplerConfig.Value.AsFloat64())
+}
+
+func getCollectorEndpoint(collectorEndpointConfig attribute.KeyValue) string {
```

Comment:
these following function are doing exactly the same - can't we make them generic?

Reply:
Done